ut_core.py

```diff
@@ -0,0 +1,149 @@
+            console: The console session object to communicate with the UTcore system.
+            test_profile_url (str, optional): The URL to a test profile YAML file. Defaults to None.
+        """
+        self.session = console
+        self.menu_config = self.load_yaml(menu_config_path)
+        self.shell = self.session.open_interactive_shell()
+
+        if test_profile_url:
+            self.test_profile = self.load_yaml(test_profile_url)
+        else:
+            self.test_profile = None
+
+    
+    def load_yaml(self, path):
+        """
+        Loads and parses a YAML file from the specified path.
+
+        Args:
+            path (str): The file path to the YAML file.
+
+        Returns:
+            dict: The parsed YAML content.
+        """
+        with open(path, 'r') as file:
+            return yaml.safe_load(file)
+
+    
+    def navigate_to_test(self, group_name, test_name):
+        """
+        Navigates the menu system to locate the specified test case.
+
+        Args:
+            group_name (str): The name of the test group to navigate to.
+            test_name (str): The name of the specific test case to locate.
+
+        Returns:
+            str: The full output from the system after navigation and test execution.
+
+        Raises:
+            ValueError: If the specified group or test is not found in the menu configuration.
+        """
+        group_found = False
+        test_found = False
+
+        # Navigate to the correct group
+        for group in self.menu_config['tvsettings']['control']['menu']['groups']:
+            if group['name'] == group_name:
+                group_found = True
+                print(f"Found group: {group_name}")
+                break
+
+        if not group_found:
+            raise ValueError(f"Group '{group_name}' not found in menu configuration.")
+
+        # Navigate to the correct test
+        for test in group['tests']:
+            if test == test_name:
+                test_found = True
+                print(f"Found test: {test_name}")
+                break
+
+        if not test_found:
+            raise ValueError(f"Test '{test_name}' not found in group '{group_name}'.")
+
+        self.session.write_to_shell("/home/FKC01/rdk-halif-power_manager/ut/bin/run.sh")
+        self.session.write_to_shell("s",3)
+        self.session.write_to_shell("1", 3)
+        self.session.write_to_shell("s", 3)
+        self.session.write_to_shell("1", 3)
+
+        full_output = self.session.get_full_output()
+
+        return full_output
+
+
+    def collect_results(self, output):
+        """
+        Collects and interprets the results from the test execution output.
+
+        Args:
+            output (str): The output from the test execution.
+
+        Returns:
+            bool: True if the test passed successfully, False if the test failed, or None if the output format is unexpected.
+        """
```

Comment:
I'll be honest I hadn't thought about this, we should check with Gerald to see if this is how he was expecting it to be done, or if he needs it to be pass/fail.